test/integration/cx_limit_overload_integration_test.cc

```diff
@@ -0,0 +1,213 @@
+  initializeOverloadManager(6);
+  std::vector<IntegrationTcpClientPtr> tcp_clients;
+  std::vector<FakeRawConnectionPtr> raw_conns;
+  for (int i = 0; i <= 5; ++i) {
+    tcp_clients.emplace_back(makeTcpConnection(lookupPort("listener_0")));
+    raw_conns.emplace_back();
+    ASSERT_TRUE(fake_upstreams_[0]->waitForRawConnection(raw_conns.back()));
+    ASSERT_TRUE(tcp_clients.back()->connected());
+  }
+  test_server_->waitForCounterEq(counterPrefix() + "downstream_global_cx_overflow", 0);
+  // 7th connection should fail because we have hit the configured limit for
+  // `max_active_downstream_connections`.
+  tcp_clients.emplace_back(makeTcpConnection(lookupPort("listener_0")));
+  raw_conns.emplace_back();
+  ASSERT_FALSE(
+      fake_upstreams_[0]->waitForRawConnection(raw_conns.back(), std::chrono::milliseconds(500)));
+  tcp_clients.back()->waitForDisconnect();
+  test_server_->waitForCounterEq(counterPrefix() + "downstream_global_cx_overflow", 1);
+  // Get rid of the client that failed to connect.
+  tcp_clients.back()->close();
+  tcp_clients.pop_back();
+  raw_conns.pop_back();
+  // Get rid of the first successfully connected client to be able to establish new connection.
+  tcp_clients.front()->close();
+  ASSERT_TRUE(raw_conns.front()->waitForDisconnect());
+  ASSERT_TRUE(waitForConnections(5));
+  // As 6th client disconnected, we can again establish a connection.
+  tcp_clients.emplace_back(makeTcpConnection(lookupPort("listener_0")));
+  raw_conns.emplace_back();
+  ASSERT_TRUE(fake_upstreams_[0]->waitForRawConnection(raw_conns.back()));
+  ASSERT_TRUE(tcp_clients.back()->connected());
+  for (auto& tcp_client : tcp_clients) {
+    tcp_client->close();
+  }
+  tcp_clients.clear();
+  raw_conns.clear();
+}
+
+TEST_P(GlobalDownstreamCxLimitIntegrationTest, GlobalLimitSetViaRuntimeKeyAndOverloadManager) {
+  // Configure global connections limit via deprecated runtime key.
+  config_helper_.addRuntimeOverride("overload.global_downstream_max_connections", "3");
+  initializeOverloadManager(2);
+  const std::string log_line =
+      "Global downstream connections limits is configured via deprecated runtime key "
+      "overload.global_downstream_max_connections and in "
+      "envoy.resource_monitors.global_downstream_max_connections. Using overload manager config.";
+  std::vector<IntegrationTcpClientPtr> tcp_clients;
+  std::vector<FakeRawConnectionPtr> raw_conns;
+  std::function<void()> establish_conns = [this, &tcp_clients, &raw_conns]() {
+    for (int i = 0; i < 2; ++i) {
+      tcp_clients.emplace_back(makeTcpConnection(lookupPort("listener_0")));
+      raw_conns.emplace_back();
+      ASSERT_TRUE(fake_upstreams_[0]->waitForRawConnection(raw_conns.back()));
+      ASSERT_TRUE(tcp_clients.back()->connected());
+    }
+  };
+  // TODO(nezdolik) this should be logged once per each run (ipv4,ipv6),
+  //  but second run does not contain the log line even though logging code is hit. Some macro+test
+  //  setup weirdness.
```

Comment:
Yea I think it is as you pointed out due to the local static in the macro that this will just be logged once for the test. 

I think we could  run the test suite with just a IP version as it shouldn't be depedent on the IP version? (as one workaround).